source/core/Array.cpp

```diff
@@ -1211,8 +1211,11 @@ VIREO_FUNCTION_SIGNATURE7(ArrayDeleteND, TypedArrayCoreRef, StaticType, void,
 
     IntIndex startIndex = offset > 0 ? offset : 0;
```

Comment:
Looking at the CG prim documentation, it looks like we may have to compare the absolute value of offset (if offset is negative) against the length and make some decisions.  Specifically, the following two scenarios:

1) If you wire a value to length and a negative value to the index, deleted portion is an array with an outer dimension size of the value of length minus the absolute value of the index. 
2) If the index is a negative value and the value of length is smaller than the absolute value of the index, deleted portion is empty.
I would recommend writing test cases for these in the VI test and ensure Vireo produces similar results as Desktop. This may be a corner case but it is likely easier to fix it now than come back to this later.

Reply:
It actually already works like that. In Desktop if you wired in -1 as the index and a 2 as the length, then it will delete the first element of the array (i.e. it sets the startIndex to 0 and the endIndex to offset {-1} + length {2 = 1).  This works both in ArrayDeleteND and ArrayDelete.

Reply:
oh ok, thanks. If we have test cases that cover these two scenarios, then we are good.